### PR TITLE
NotificationMailer setup

### PR DESCRIPTION
**Pull request by AnkushMalik**
In this PR: 
- NotificationMailer Added
- Add crono gem and NotificationJob (to send mails to specific users)
- some tests
___

image

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -61,4 +61,18 @@
 
   # Enable authentification test mode
   config.devise.ichain_test_mode = true
+
+  config.action_mailer.delivery_method = :smtp
+
+  config.action_mailer.default_url_options = { host: 'localhost:3000' }
+
+  config.action_mailer.smtp_settings = {
```
[the remark this answers is not in the document]
Comment (AnkushMalik) [reply config/environments/development.rb]: you mean to set it up as ENV variable?

File: db/schema.rb [hunk 2]
```diff
@@ -55,46 +65,20 @@
     t.index ["project_id"], name: "index_episodes_projects_on_project_id"
   end
 
-  create_table "impressions", id: :integer, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci", force: :cascade do |t|
```
[the remark this answers is not in the document]
Comment (AnkushMalik) [reply db/schema.rb]: bad git 😅

File: config/environments/production.rb [hunk 3]
```diff
@@ -107,4 +107,18 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  config.action_mailer.default_url_options = { host: 'hackweek.suse.com' }
+
+  config.action_mailer.delivery_method = :smtp
+
+  config.action_mailer.smtp_settings = {
```
[the remark this answers is not in the document]
Comment (AnkushMalik) [reply config/environments/production.rb]: smtp?

**Comment by differentreality** [comment]
@AnkushMalik  the point about the configuration settings is that if I want to clone this project and run it on my own server, I should be able to define those settings without making changes in the codebase itself. I may or may not use gmail, for example. I obviously won't be using hackweek.suse.com as host etc etc. 
Makes sense?

File: config/environments/development.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (hennevogel) [reply config/environments/development.rb]: Yes in `application.yml`

File: config/environments/production.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (hennevogel) [reply config/environments/production.rb]: All of it.

File: config/cronotab.rb [hunk 4]
```diff
@@ -0,0 +1 @@
+Crono.perform(NotificationJob).every 1.day, at: '00:00'
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply config/cronotab.rb]: The later sounds sane...

**Comment by hennevogel** [comment]
@AnkushMalik are you going to continue this?

**Pull request closed without merging**